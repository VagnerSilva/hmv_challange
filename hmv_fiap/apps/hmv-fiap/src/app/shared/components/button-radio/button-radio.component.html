<div
	class="my-4"
	[formGroup]="parentFormGroup.form"
	class="input-container"
	(focusin)="onFocusIn($event)"
	(focusout)="onFocusOut($event)"
>
	<label>{{ labelName }}</label>
	<mat-radio-group class="block" [formControl]="$any(ngControl.control)">
		<mat-radio-button
			*ngFor="let item of options"
			class="font-bold uppercase radio-space"
			[value]="item[keyValue] ? item[keyValue] : item"
		>
			{{ item[keyLabel] ? item[keyLabel] : item }}
		</mat-radio-button>
	</mat-radio-group>
	<ng-container *ngIf="ngControl.control?.touched">
		<mat-error
			class="form-text text-danger"
			*ngFor="
				let errorMessage of $any(parentFormGroup.form.controls)[$any(ngControl.name)][
					'errorMessages'
				]
			"
		>
			{{ errorMessage }}<br />
		</mat-error>
	</ng-container>
</div>
