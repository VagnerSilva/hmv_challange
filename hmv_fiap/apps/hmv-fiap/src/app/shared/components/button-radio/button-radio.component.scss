.radio-space:not(:first-child) {
	margin-left: 56px;
}
