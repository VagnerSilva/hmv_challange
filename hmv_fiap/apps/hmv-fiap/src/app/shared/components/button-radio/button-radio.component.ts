import {
	Attribute,
	ChangeDetectionStrategy,
	ChangeDetectorRef,
	Component,
	ElementRef,
	Input,
	Optional,
	Self,
	ViewEncapsulation,
} from '@angular/core'
import { FormGroupDirective, NgControl, NgForm } from '@angular/forms'
import { ErrorStateMatcher } from '@angular/material/core'
import { InputBaseComponent } from '../utils/input-base.component'

@Component({
	selector: 'hmv-button-radio',
	templateUrl: './button-radio.component.html',
	styleUrls: ['./button-radio.component.scss'],
	encapsulation: ViewEncapsulation.None,
	changeDetection: ChangeDetectionStrategy.OnPush,
})
export class ButtonRadioComponent extends InputBaseComponent<ButtonRadioComponent> {
	@Input() options: any[] = []
	constructor(
		@Attribute(`labelName`) public labelName: string,
		@Attribute(`keyValue`) public keyValue: string,
		@Attribute(`keyLabel`) public keyLabel: string,
		@Optional() @Self() public override readonly ngControl: NgControl,
		@Optional() public override readonly parentFormGroup: FormGroupDirective,
		@Optional() public override readonly parentForm: NgForm,
		public readonly elementRef: ElementRef,
		public override readonly defaultErrorStateMatcher: ErrorStateMatcher,
		public override cdr: ChangeDetectorRef,
	) {
		super(elementRef, parentFormGroup, parentForm, defaultErrorStateMatcher, ngControl, cdr)
	}
}
