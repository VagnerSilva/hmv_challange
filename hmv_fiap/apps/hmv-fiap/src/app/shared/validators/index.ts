export class Validator {
	static msg = new Validator()

	REQUIRED = 'Campo obrigatório'

	MIN(value: number): string {
		return `Campo deve ter no mínimo ${value} caracteres`
	}
}
