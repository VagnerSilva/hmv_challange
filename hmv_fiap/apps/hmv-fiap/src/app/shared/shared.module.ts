import { CommonModule } from '@angular/common'
import { NgModule } from '@angular/core'
import { ReactiveFormsModule } from '@angular/forms'
import { RxReactiveFormsModule } from '@rxweb/reactive-form-validators'
import { ButtonRadioComponent } from './components/button-radio/button-radio.component'
import { ButtonComponent } from './components/button/button.component'
import { InputComponent } from './components/input/input.component'
import { SelectComponent } from './components/select/select.component'
import { TextareaComponent } from './components/textarea/textarea.component'
import { MaterialModule } from './material/material.module'

const components = [
	ButtonComponent,
	InputComponent,
	SelectComponent,
	TextareaComponent,
	ButtonRadioComponent,
]

const modules = [MaterialModule, ReactiveFormsModule]

@NgModule({
	declarations: [...components],
	imports: [CommonModule, ...modules],
	exports: [...components, ...modules, RxReactiveFormsModule],
})
export class SharedModule {}
