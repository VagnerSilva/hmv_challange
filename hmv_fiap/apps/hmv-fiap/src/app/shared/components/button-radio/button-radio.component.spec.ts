import { Component } from '@angular/core'
import { FormBuilder, FormGroup, NgForm } from '@angular/forms'
import { createHostFactory, SpectatorHost } from '@ngneat/spectator/jest'
import { SharedModule } from '@shared/shared.module'
import { ButtonRadioComponent } from './button-radio.component'

const cases = {
	simples: ` <form [formGroup]="formExample">
  <hmv-button-radio formControlName="select" labelName="My Radio" [options]="options"></hmv-button-radio>`,
}

@Component({ template: '' })
class CustomHostComponent {
	formExample!: FormGroup
	options = ['Sim', 'Nao']

	constructor(private fb: FormBuilder) {
		this.formExample = this.fb.group({
			select: ['', []],
		})
	}
	title = 'Custom HostComponent'
}

describe('ButtonRadioComponent', () => {
	let host: SpectatorHost<ButtonRadioComponent, CustomHostComponent>
	const createHost = createHostFactory({
		component: ButtonRadioComponent,
		host: CustomHostComponent,
		imports: [SharedModule],
		mocks: [NgForm],
	})

	it('should create', () => {
		host = createHost(cases.simples)

		expect(host.component).toBeTruthy()
		expect(host.fixture).toMatchSnapshot()
	})

	it('should select a option', () => {
		host = createHost(cases.simples)
		const check = host.query('input[value="Sim"]') as HTMLElement
		const getSelect = () => host.hostComponent.formExample.value

		check.click()
		expect(getSelect().select).toBe('Sim')
		const check2 = host.query('input[value="Nao"]') as HTMLElement
		check2.click()
		expect(getSelect().select).not.toBe('Sim')
	})
})
