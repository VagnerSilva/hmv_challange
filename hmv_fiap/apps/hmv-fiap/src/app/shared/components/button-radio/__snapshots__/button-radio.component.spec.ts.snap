// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonRadioComponent should create 1`] = `
<ng-component
  __ngContext__={[Function LRootView]}
  fb={[Function FormBuilder]}
  formExample={[Function FormGroup]}
  options={[Function Array]}
  title={[Function String]}
>
  <form
    class="ng-untouched ng-pristine ng-valid"
    novalidate=""
  >
    <hmv-button-radio
      class="ng-untouched ng-pristine ng-valid"
      formcontrolname="select"
      id="input-0"
      labelname="My Radio"
    >
      <div
        class="input-container ng-untouched ng-pristine ng-valid"
      >
        <label>
          My Radio
        </label>
        <mat-radio-group
          class="mat-radio-group block ng-untouched ng-pristine ng-valid"
          role="radiogroup"
        >
          <mat-radio-button
            class="mat-radio-button font-bold uppercase radio-space _mat-animation-noopable mat-accent"
          >
            <label
              class="mat-radio-label"
              for="mat-radio-2-input"
            >
              <span
                class="mat-radio-container"
              >
                <span
                  class="mat-radio-outer-circle"
                />
                <span
                  class="mat-radio-inner-circle"
                />
                <input
                  class="mat-radio-input cdk-visually-hidden"
                  id="mat-radio-2-input"
                  name="mat-radio-group-0"
                  tabindex="0"
                  type="radio"
                  value="Sim"
                />
                <span
                  class="mat-ripple mat-radio-ripple mat-focus-indicator"
                  mat-ripple=""
                >
                  <span
                    class="mat-ripple-element mat-radio-persistent-ripple"
                  />
                </span>
              </span>
              <span
                class="mat-radio-label-content"
              >
                <span
                  style="display: none;"
                >
                </span>
                 Sim 
              </span>
            </label>
          </mat-radio-button>
          <mat-radio-button
            class="mat-radio-button font-bold uppercase radio-space _mat-animation-noopable mat-accent"
          >
            <label
              class="mat-radio-label"
              for="mat-radio-3-input"
            >
              <span
                class="mat-radio-container"
              >
                <span
                  class="mat-radio-outer-circle"
                />
                <span
                  class="mat-radio-inner-circle"
                />
                <input
                  class="mat-radio-input cdk-visually-hidden"
                  id="mat-radio-3-input"
                  name="mat-radio-group-0"
                  tabindex="0"
                  type="radio"
                  value="Nao"
                />
                <span
                  class="mat-ripple mat-radio-ripple mat-focus-indicator"
                  mat-ripple=""
                >
                  <span
                    class="mat-ripple-element mat-radio-persistent-ripple"
                  />
                </span>
              </span>
              <span
                class="mat-radio-label-content"
              >
                <span
                  style="display: none;"
                >
                </span>
                 Nao 
              </span>
            </label>
          </mat-radio-button>
        </mat-radio-group>
      </div>
    </hmv-button-radio>
  </form>
</ng-component>
`;
